### PR TITLE
prompt to input user/pass when missing

### PR DESCRIPTION
**Pull request by yuex**
if no username or password is pre-configured, will prompt user to input. And the authentication moved in simplenote#SimpleNote(), thus the plugin won't explode if you interrupt your first login and want to login again or your login failed.

### CODE REVIEW

**Comment by atomicules** [comment]
Hi. Thanks for this. I will review it and update back here in a couple of  days time.

File: autoload/simplenote.vim [hunk 1]
```diff
@@ -66,13 +66,13 @@ else
...
-  echoerr errmsg
-  finish
-endif
+" if (s:user == "") || (s:password == "")
+  " let errmsg = "Simplenote credentials missing. Set g:SimplenoteUsername and "
+  " let errmsg = errmsg . "g:SimplenotePassword. If you don't have an account you can "
+  " let errmsg = errmsg . "create one at https://simple-note.appspot.com/create/."
+  " echoerr errmsg
+  " finish
+" endif
```
Comment (m1foley) [inline autoload/simplenote.vim]: It looks like you left in some commented-out code.

**Comment by atomicules** [comment]
Sorry, will probably be the weekend now when I get around to reviewing this.

File: autoload/simplenote.vim [hunk 2]
```diff
@@ -728,49 +723,75 @@ endfunction
...
-            datetime.datetime.strptime(vim.eval("a:1"), "%Y-%m-%d")
-            interface.list_note_index_in_scratch_buffer(since=vim.eval("a:1"))
-        except ValueError:
-            interface.list_note_index_in_scratch_buffer(tags=vim.eval("a:1").split(","))
-    else:
-        interface.list_note_index_in_scratch_buffer()
-
-elif param == "-d":
-    interface.trash_current_note()
+            vim.command("let s:user=input('username:', '')")
```
Comment (m1foley) [inline autoload/simplenote.vim]: Simplenote calls this "Email" on their website & app, so I think that should be used instead of "username".

**Comment by m1foley** [comment]
I just tested this and it works as expected.

**Comment by atomicules** [comment]
Thanks for testing. I still need to make some tweaks to either my 
Vader tests or this pull request though so my tests still run ok.

I just don't have the time in the evenings this week. Will be able 
to get around to it at the weekend though.

**Comment by atomicules** [comment]
Hi. So I did get around to looking at this over the weekend. I can't merge it as is, because it breaks current functionality, it seems to reset/ignore the `g:SimplenoteUsername` after a point.

However, I do like the idea and the idea of using `SimplenoteLoginFailed` so I'm going to carry on working on this to get it to something that can be merged in, likely to be anywhere between a few days to a few weeks.

When entering a username/password, how were you intending logon failures to work? Do you want to be continually prompted for username/password until it is correct? The user could always CTRL+C out of it.

**Comment by yuex** [comment]
If just one of `g:SimplenoteUsername` and `g:SimplenotePassword` is set, both will be ignored. But not if both are set. You mean there's bug about this logic?

In the PR, when logon failed, a warning will be printed, and the function will return. The user is free to launch the command and try again

**Comment by atomicules** [comment]
Hi. I've just pushed some minor tweaks to your pull request to a 
temporary branch, just so you have a chance to review before I close 
this out. The changes I made are explained in the commit message and I 
tried to keep as much as your code as possible since you've gone to 
the trouble to write it.

The only thing I might think about before closing this out is if there 
is anyway to prevent the Simplenote buffer from being created if a 
login exception occurs. I don't think there is a clean way though as 
the buffer is created first. It's not a huge deal either.

Oh, I'll also have to update the README before closing out to mention 
this new functionality.

**Comment by atomicules** [comment]
I think this is pretty much ready to close out. I've had a look into 
the buffer thing and I think what I really need to do is to change 
simplenote.py so during initialisation it tries to fetch a token 
straight away. That way we can catch the SimplenoteLoginFailed as 
early as possible (i.e. in initialisation of SimplenoteVimInterface). 
But that's a job for another day.

**Comment by yuex** [comment]
looks good to me. I close this PR now and feel free to re-open it

**Comment by atomicules** [comment]
I've merged all that into master now. Thanks for your contribution!

**Pull request closed without merging**